editor/src/components/canvas/canvas-strategies/draw-to-insert-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,766 @@
+import { act, fireEvent } from '@testing-library/react'
+import { jsxElement, simpleAttribute } from '../../../core/shared/element-template'
+import { windowPoint, WindowPoint } from '../../../core/shared/math-utils'
+import { EditorAction } from '../../editor/action-types'
+import { enableInsertModeForJSXElement } from '../../editor/actions/action-creators'
+import {
+  makeTestProjectCodeWithSnippet,
+  renderTestEditorWithCode,
+  getPrintedUiJsCode,
+} from '../ui-jsx.test-utils'
+import { CanvasControlsContainerID } from '../controls/new-canvas-controls'
+import { wait } from '../../../utils/utils.test-utils'
+
+function slightlyOffsetWindowPointBecauseVeryWeirdIssue(point: {
+  x: number
+  y: number
+}): WindowPoint {
+  // FIXME when running in headless chrome, the result of getBoundingClientRect will be slightly
+  // offset for some unknown reason, meaning the inserted element will be 1 pixel of in each dimension
```

Comment:
👀

Reply:
I haven't touched this, this is just renaming of insert-mode.spec.browser2.tsx and I added new test cases